front/src/common/authorization/hooks/useUserResourceGrants.ts

```diff
@@ -0,0 +1,62 @@
+import { useEffect } from 'react';
+
+import {
+  mockedEditoastApi,
+  type GetGrantsByResourceTypeResponse,
+  type ResourceType,
+} from 'common/api/mock/mockEditoastApi';
+
+type ResourceGrants = Partial<Record<ResourceType, GetGrantsByResourceTypeResponse>>;
```

Comment:
nit : We sometimes put an `s` to resource, sometimes not. Looks like this hook can manipulate until 2 resources (infra and timetable) so we should maybe put an `s` everywhere ? 
```suggestion
type ResourcesGrants = Partial<Record<ResourceType, GetGrantsByResourceTypeResponse>>;
```